allennlp/training/trainer.py

```diff
@@ -31,6 +31,24 @@
 logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
 
+class TensorboardWriter:
+    """
+    Wraps a pair of ``SummaryWriter`` instances but is a no-op if they're ``None``.
+    Allows Tensorboard logging without always checking for Nones first.
+    """
+    def __init__(self, train_log: SummaryWriter = None, validation_log: SummaryWriter = None) -> None:
+        self._train_log = train_log
+        self._validation_log = validation_log
+
+    def add_train_scalar(self, name: str, value: float, global_step=None) -> None:
+        if self._train_log is not None:
+            self._train_log.add_scalar(name, value, global_step)
+
+    def add_validation_scalar(self, name: str, value: float, global_step=None) -> None:
```

Comment:
Why is global_step optional?

Reply:
I'm sure I must have had a good reason for it, but I couldn't tell you what it is now.

Reply:
I'll make it non-optional.